crates/classification-macros/src/lib.rs

```diff
@@ -103,15 +103,9 @@ pub fn classify_bytes_derive(input: TokenStream) -> TokenStream {
             ];
         }
 
-        // Extend u8 with a `classify` method
-        trait U8Ext {
-            fn classify(&self) -> #enum_name;
-        }
-
-        impl U8Ext for u8 {
-            #[inline(always)]
-            fn classify(&self) -> #enum_name {
-                #enum_name::TABLE[*self as usize]
+        impl From<u8> for #enum_name {
+            fn from(byte: u8) -> Self {
```

Comment:
I guess this will always be inlined by Rust just b/c small (Rust v1.75+ change)?

Reply:
Didn't check but also didn't notice any perf impact at all.